atc/worker/client.go

```diff
@@ -427,6 +430,8 @@ func (client *client) RunGetStep(
 		return GetResult{}, err
 	}
 
+	eventDelegate.SelectedWorker(logger, chosenWorker.Name())
```

Comment:
minor: Might make sense to put this just below the Signature check, as if that fails, this info won't be useful

Reply:
My idea was to emit worker info as early as possible. Because if everything goes fine, then we don't need to check worker name at all; but if something goes wrong, any error might be related to worker.